src/ImageCarouselReact/ImageCarouselReact.xml

```diff
@@ -0,0 +1,211 @@
+                    </property>
+
+                    <property key="onClickEvent" type="enumeration" defaultValue="non">
+                        <caption>On Click</caption>
+                        <category>Behavior</category>
+                        <description>Location to open the form, either content, popup or Blocking Popup </description>
+                        <enumerationValues>
+                            <enumerationValue key="non">Do Nothing</enumerationValue>
+                            <enumerationValue key="microflow">Call Mircoflow</enumerationValue>
+                            <enumerationValue key="content">Open Page</enumerationValue>
+                            <enumerationValue key="popup">Popup</enumerationValue>
+                            <enumerationValue key="modal">Blocking Popup</enumerationValue>
+                        </enumerationValues> 
+                    </property>
+
+                     <property key="imageClickMicroflow" type="microflow" required="false">
+                        <caption>Call Microflow</caption>
+                        <category>Behavior</category>
+                        <description>Microflow to be triggered when an image is clicked.</description>
+                        <returnType type="Void"/>
+                    </property>
+
+                    <property key="openPage" type="form" required="false">
+                        <caption>Open Page</caption>
+                        <category>Behavior</category>
+                        <description>Opens a page when Image is Clicked</description> 
+                    </property>
+
+                </properties>
+            </property>
+            <property key="captionAttr" type="attribute" entityProperty="imageEntity" required="false">
+                <caption>Caption</caption>
+                <category>Data source</category>
+                <description>This fields provides the caption of a carousel item</description> 
+                <attributeTypes>
+                     <attributeType name="String"/> 
+                </attributeTypes>
+            </property>
+
+            <property key="descriptionAttr" type="attribute" entityProperty="imageEntity" required="false">
+                <caption>Description</caption>
+                <category>Data source</category>
+                <description>This fields provides the Description of a carousel item</description> 
+                <attributeTypes>
+                     <attributeType name="String"/> 
+                </attributeTypes>
+            </property>
+            
+            <property key="controls" type="boolean" defaultValue="true">
+                <caption>Use Image Navigation</caption>
+                <category>Carousel</category>
+                <description>Use the image navigation.</description>
+            </property>
+
+            <property key="indicators" type="boolean" defaultValue="true">
```

Comment:
Its a user's preference.. maybe he/she just wants to see the images without anything on the images.  I have removed this for now, it can be added at a later state